index.js

```diff
@@ -78,9 +81,28 @@ const portCheckSequence = function * (ports) {
 
 export default async function getPorts(options) {
 	let ports;
+	let exclude = new Set();
 
 	if (options) {
-		ports = typeof options.port === 'number' ? [options.port] : options.port;
+		if (options.port) {
+			ports = typeof options.port === 'number' ? [options.port] : options.port;
+		}
+
+		if (options.exclude) {
+			const excludeIterable = options.exclude;
+
+			if (typeof excludeIterable[Symbol.iterator] !== 'function') {
+				throw new TypeError('The `exclude` option must be an iterable.');
+			}
+
+			for (const element of excludeIterable) {
+				if (typeof element !== 'number') {
```

Comment:
I would use `Number.isSafeInteger` here.

Reply:
Never knew that existed, thanks for the insight, I'll add that!